addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -181,9 +187,27 @@ def write(self, vals):
                     raise AccessError(_('You can not write on %(field_name)s.', field_name=field_name))
         return super().write(vals)
 
+    @api.autovacuum
+    def _gc_unpin_outdated_sub_channels(self):
+        members = self.env["discuss.channel.member"].search(
+            [
+                ("channel_id.parent_channel_id", "!=", False),
+                ("last_interest_dt", "<", fields.Datetime.now() - timedelta(days=2)),
+            ]
+        )
+        members.unpin_dt = fields.Datetime.now()
+        for member in members:
+            member._bus_send("discuss.channel/unpin", {"id": member.channel_id.id})
+
     def unlink(self):
         # sudo: discuss.channel.rtc.session - cascade unlink of sessions for self member
         self.sudo().rtc_session_ids.unlink()  # ensure unlink overrides are applied
+        # Always unlink members of sub-channels as well. Member list should be
+        # kept in sync.
+        for member in self.channel_id.sub_channel_ids.channel_member_ids.filtered(
+            lambda m: m.partner_id in self.partner_id or m.guest_id in self.guest_id
+        ):
```

Comment:
If there are many sub-channels or sub-channel members involved, search might be better than using field and python filtering.